### PR TITLE
[feature](Azure) Support copy into on Azure Blob Storage

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

Support copy into logic in FE on Azure Blob Storage.

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/cloud/storage/AzureRemote.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+            }
+        } catch (BlobStorageException e) {
+            LOG.warn("Failed to delete objects for Azure", e);
+            throw new DdlException("Failed to delete objects for Azure, Error message=" + e.getMessage());
+        }
+    }
+
+    @Override
+    public void close() {
+        if (client != null) {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/storage/AzureRemote.java:195]: Redundant check

File: fe/fe-core/src/main/java/org/apache/doris/cloud/storage/AzureRemote.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    private void initClient() {
+        if (client == null) {
+            BlobContainerClientBuilder builder = new BlobContainerClientBuilder();
+            if (obj.getToken() != null) {
+                builder.credential(new SimpleTokenCredential(obj.getToken()));
+            } else {
+                builder.credential(new StorageSharedKeyCredential(obj.getAk(), obj.getSk()));
+            }
+            String containerName = obj.getBucket();
+            String uri = String.format("https://%s.blob.core.windows.net/%s", obj.getAk(),
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/storage/AzureRemote.java:234]: constant

File: fe/fe-core/src/main/java/org/apache/doris/cloud/storage/AzureRemote.java [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+        }
+    }
+
+    @Override
+    public Triple<String, String, String> getStsToken() throws DdlException {
+        try {
+            BlobContainerClientBuilder builder = new BlobContainerClientBuilder();
+            builder.credential(new StorageSharedKeyCredential(obj.getAk(), obj.getSk()));
+            String containerName = obj.getBucket();
+            String uri = String.format("https://%s.blob.core.windows.net/%s", obj.getAk(),
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/storage/AzureRemote.java:137]: constant

File: fe/fe-core/src/main/java/org/apache/doris/cloud/storage/AzureRemote.java [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+
+    private static final Logger LOG = LogManager.getLogger(DefaultRemote.class);
+
+    private BlobContainerClient client;
+
+    public AzureRemote(ObjectInfo obj) {
+        super(obj);
+    }
+
+    private long SESSION_EXPIRE_SECOND = 1800;
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/storage/AzureRemote.java:67]: Make sure this value is the same as other implementations of RemoteBase such as ObsRemote and OssRemote.

File: fe/fe-core/src/main/java/org/apache/doris/cloud/storage/AzureRemote.java [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+import org.apache.logging.log4j.Logger;
+import reactor.core.publisher.Mono;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AzureRemote extends RemoteBase {
+
+    private static final Logger LOG = LogManager.getLogger(DefaultRemote.class);
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/storage/AzureRemote.java:59]: DefaultRemote.class -> AzureRemote.class

**Review by gavinchou — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/cloud/storage/AzureRemote.java [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+
+    private long SESSION_EXPIRE_SECOND = 1800;
+
+    @Override
+    public String getPresignedUrl(String fileName) {
+        try {
+            BlobContainerClientBuilder builder = new BlobContainerClientBuilder();
+            builder.credential(new StorageSharedKeyCredential(obj.getAk(), obj.getSk()));
+            String containerName = obj.getBucket();
+            String uri = String.format("https://%s.blob.core.windows.net/%s", obj.getAk(),
```
[the remark this answers is not in the document]
Comment (ByteYue) [reply fe/fe-core/src/main/java/org/apache/doris/cloud/storage/AzureRemote.java:75]: I wonder should we make the scheme about whether use http or https a config?

File: fe/fe-core/src/main/java/org/apache/doris/cloud/storage/AzureRemote.java (hunk 1 above)
Comment (ByteYue) [reply fe/fe-core/src/main/java/org/apache/doris/cloud/storage/AzureRemote.java:195]: Why? DefaultRemote does the same check.

**Review by gavinchou — DISMISSED** [review]

**Review by liaoxin01 — DISMISSED** [review]

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by ByteYue** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by liaoxin01 — APPROVED** [review]

**Pull request merged**